central/networkgraph/entity/datastore/internal/store/postgres/full_store.go

```diff
@@ -0,0 +1,85 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/stackrox/rox/central/metrics"
+	"github.com/stackrox/rox/central/networkgraph/entity/datastore/internal/store"
+	"github.com/stackrox/rox/pkg/env"
+	ops "github.com/stackrox/rox/pkg/metrics"
+	"github.com/stackrox/rox/pkg/postgres"
+	pkgSchema "github.com/stackrox/rox/pkg/postgres/schema"
+)
+
+const (
+	networkFlowsTable    = pkgSchema.NetworkFlowsTableName
+	networkEntitiesTable = pkgSchema.NetworkEntitiesTableName
+
+	pruneOrphanExternalNetworkEntitiesSrcStmt = `DELETE FROM %s entity WHERE NOT EXISTS
+		(SELECT 1 FROM %s flow WHERE flow.Props_SrcEntity_Type = 4
+		AND flow.Props_SrcEntity_Id = entity.Info_Id
+		AND entity.Info_ExternalSource_Discovered = true);`
+
+	pruneOrphanExternalNetworkEntitiesDstStmt = `DELETE FROM %s entity WHERE NOT EXISTS
+		(SELECT 1 FROM %s flow WHERE flow.Props_DstEntity_Type = 4
+		AND flow.Props_DstEntity_Id = entity.Info_Id
+		AND entity.Info_ExternalSource_Discovered = true);`
+)
+
+var (
+	queryTimeout = env.PostgresDefaultNetworkFlowQueryTimeout.DurationSetting()
+)
+
+// NewFullStore augments the generated store with RemoveOrphanedEntities function.
```

Comment:
[nit] could we expand this comment to describe why this is necessary? my understanding is that it is due to the use of the generic store in `store.go` which can't otherwise be extended (is that correct?)

Reply:
There is maybe an argument that we don't actually need this distinction between the store and full store, since we've modified the `EntityStore` interface with the new method. Perhaps we can simply modify the existing `New` function? WDYT?

Reply:
As I understand it, central/networkgraph/entity/datastore/internal/store/postgres/store.go is entirely generated, so there is no way we modify the content of it. The `full_store` expansion trick seems to be the usual technique to add additional functionality to the generated store.
An example is : central/deployment/datastore/internal/store/postgres/full_store.go